src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/ChunkCrossOvers.java

```diff
@@ -0,0 +1,204 @@
+    for (int i = 0; i < fromChunkOffset; i++) {
+      fromIterator.next();
+    }
+    // As keys can not be overwritten, only removed and new ones added, this
+    // cross over overwrites the values. Removal of keys is handled by the
+    // removeChunk mutation. Value equality is not checked here.
+    for (int i = 0; i < chunkSize; i++) {
+      Entry<K, V> from = fromIterator.next();
+      Entry<K, V> to = toIterator.next();
+      to.setValue(from.getValue());
+    }
+  }
+
+  static <K, V> void crossOverChunk(Map<K, V> map, Map<K, V> otherMap,
+      SerializingMutator<K> keyMutator, SerializingMutator<V> valueMutator, PseudoRandom prng) {
+    int maxChunkSize = Math.min(map.size(), otherMap.size());
+    int chunkSize = prng.closedRangeBiasedTowardsSmall(1, maxChunkSize);
+    int fromChunkOffset = prng.closedRange(0, otherMap.size() - chunkSize);
+    int toChunkOffset = prng.closedRange(0, map.size() - chunkSize);
+
+    Iterator<Entry<K, V>> toIterator = map.entrySet().iterator();
+    for (int i = 0; i < toChunkOffset; i++) {
+      toIterator.next();
+    }
+    Iterator<Entry<K, V>> fromIterator = otherMap.entrySet().iterator();
+    for (int i = 0; i < fromChunkOffset; i++) {
+      fromIterator.next();
+    }
+
+    // Remove keys only after all cross-overs to prevent generating them anew.
+    List<K> keysToRemove = new ArrayList<>(chunkSize);
+    List<Map.Entry<K, V>> entriesToAdd = new ArrayList<>(chunkSize);
+
+    // If a key cross over results in an already existing key or null, it is
+    // retried up to MAX_FAILED_CROSS_OVER_ATTEMPTS times. Similarly, a change
+    // of the value is attempted.
+    for (int i = 0; i < chunkSize; i++) {
+      Entry<K, V> to = toIterator.next();
+      Entry<K, V> from = fromIterator.next();
+
+      // Keys and values are still in the map and would be mutated without
+      // detaching them prior to the cross-over.
+      K detachedKey = keyMutator.detach(to.getKey());
+      K newKey = attemptCrossOver(
+          detachedKey, from.getKey(), keyMutator, prng, k -> k != null && !map.containsKey(k));
+
+      V detachedValue = valueMutator.detach(to.getValue());
+      V newValue = attemptCrossOver(
+          detachedValue, from.getValue(), valueMutator, prng, v -> !to.getValue().equals(v));
+
+      // If the key could be crossed over, replace the whole entry, otherwise
+      // only set the (potentially) new value for the existing key.
+      if (newKey != null) {
+        entriesToAdd.add(new SimpleEntry<>(newKey, newValue));
+        keysToRemove.add(to.getKey());
+      } else {
+        to.setValue(newValue);
+      }
+    }
+    keysToRemove.forEach(map::remove);
+    entriesToAdd.forEach(e -> map.put(e.getKey(), e.getValue()));
+  }
+
+  private static <T> T attemptCrossOver(T reference, T otherReference, ValueMutator<T> mutator,
+      PseudoRandom prng, Predicate<T> predicate) {
+    int remainingAttempts = MAX_FAILED_CROSS_OVER_ATTEMPTS;
```

Comment:
Reasoning was that a failed cross over, e.g. due to equal values, which could happen regularly, should not prevent the cross over of other elements in the map. To mitigate this somewhat, `MAX_FAILED_CROSS_OVER_ATTEMPTS` is only set to `10`. We could probably reduce that even further.

Reply:
My worry is that if we try to cross over two large lists that are mostly identical, the attempt would fail essentially every time. We could somewhat guard against that by checking whether `reference` and `otherReference` are equal the first time cross over fails.

Reply:
As discussed, removed the retry.